File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.py
class Solution(object):
    def smallestIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        for i in range(len(nums)):
            digit = nums[i]

            total = 0
            while(digit > 0):
                total += digit % 10
                digit //= 10
            if(total == i):
                return i
        return -1

File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.java
class Solution {
    public int smallestIndex(int[] nums) {
        for(int i = 0; i< nums.length; i++){
            int digit = nums[i];

            int total = 0;
            while(digit > 0){
                total += digit % 10;
                digit = digit / 10;
            }
            if(total == i){
                return i;
            }
        }
        return -1;
    }
}